File: .sling/libs/web5/eqela/jk-core/0.36.0/jk.log.js
let JkLogLoggingContext = {};

JkLogLoggingContext.IS_INSTANCE = function(o) {
	return o != null && o._t != null && o._t["JkLogLoggingContext"] === true;
};

let JkLogLog = function() {};

JkLogLog.prototype._t = { "JkLogLog" : true };
JkLogLog.prototype._tobj = JkLogLog;

JkLogLog.NEW = function() {
	var v = new JkLogLog;
	return v.CTOR_JkLogLog();
};

JkLogLog.prototype.CTOR_JkLogLog = function() {
	return this;
};

JkLogLog.asString = function(context, value) {
	if(context != null && (JkLangStringObjectWithDebug.IS_INSTANCE && JkLangStringObjectWithDebug.IS_INSTANCE(value))) {
		return value.toStringWithDebug((context.isInDebugMode()));
	}
	return JkLangString.asString(value);
};

JkLogLog.error = function(context, message) {
	if(context != null) {
		context.logError(message);
	}
};

JkLogLog.warning = function(context, message) {
	if(context != null) {
		context.logWarning(message);
	}
};

JkLogLog.info = function(context, message) {
	if(context != null) {
		context.logInfo(message);
	}
};

JkLogLog.debug = function(context, message) {
	if(context != null) {
		context.logDebug(message);
	}
};

JkLogLog.status = function(context, message) {
	if(context != null) {
		context.logStatus(message);
	}
};

JkLogLog.IS_INSTANCE = function(o) {
	return o != null && o._t != null && o._t["JkLogLog"] === true;
};

File: .sling/libs/sushi/eqela/jk-core/0.35.0/jk.log.lua
jk = jk or {}

jk.log = jk.log or {}

jk.log.Log = {}
jk.log.Log.__index = jk.log.Log
_vm:set_metatable(jk.log.Log, {})

function jk.log.Log._create()
	local v = _vm:set_metatable({}, jk.log.Log)
	return v
end

function jk.log.Log:_init()
	self._isClassInstance = true
	self._qualifiedClassName = self._qualifiedClassName or 'jk.log.Log'
	self['_isType.jk.log.Log'] = true
end

function jk.log.Log:_construct0()
	jk.log.Log._init(self)
	return self
end

function jk.log.Log:asString(context, value)
	if context ~= nil and (_vm:to_table_with_key(value, '_isType.jk.lang.StringObjectWithDebug') ~= nil) then
		do return value:toStringWithDebug(context:isInDebugMode()) end
	end
	do return _g.jk.lang.String:asString(value) end
end

function jk.log.Log:error(context, message)
	if context ~= nil then
		do context:logError(message) end
	end
end

function jk.log.Log:warning(context, message)
	if context ~= nil then
		do context:logWarning(message) end
	end
end

function jk.log.Log:info(context, message)
	if context ~= nil then
		do context:logInfo(message) end
	end
end

function jk.log.Log:debug(context, message)
	if context ~= nil then
		do context:logDebug(message) end
	end
end

function jk.log.Log:status(context, message)
	if context ~= nil then
		do context:logStatus(message) end
	end
end

jk.log.LoggingContext = {}
